component/tests/common.py

```diff
@@ -152,31 +128,13 @@ class ComponentRegistryCase(
 
     """
 
-    if sys.version_info < (3, 8):
-        # Copy/paste from
-        # https://github.com/odoo/odoo/blob/0218d870d319af4f263d5e9aa324990f7cc90139/
-        # odoo/tests/common.py#L248-L268
-        # Partial backport of bpo-24412, merged in CPython 3.8
-        _class_cleanups = []
-
-        @classmethod
-        def addClassCleanup(cls, function, *args, **kwargs):
-            """Same as addCleanup, except the cleanup items are called even if
-            setUpClass fails (unlike tearDownClass). Backport of bpo-24412."""
-            cls._class_cleanups.append((function, args, kwargs))
-
-        @classmethod
-        def doClassCleanups(cls):
-            """Execute all class cleanup functions.
-            Normally called for you after tearDownClass.
-            Backport of bpo-24412."""
-            cls.tearDown_exceptions = []
-            while cls._class_cleanups:
-                function, args, kwargs = cls._class_cleanups.pop()
-                try:
-                    function(*args, **kwargs)
-                except Exception:
-                    cls.tearDown_exceptions.append(sys.exc_info())
+    def setUp(self):
+        super().setUp()
+        ComponentRegistryCase._setup_registry(self)
```

Comment:
Again, this goes to `setUpClass`.

Reply:
So, then I can't pass tests locally.
Here what I have:
```
...
@classmethod
    def setUpClass(cls):
        super().setUpClass()
        ComponentRegistryCase._setup_registry(cls)

    @classmethod
    def tearDownClass(cls):
        super().tearDownClass()
        ComponentRegistryCase._teardown_registry(cls)

    @staticmethod
    def _setup_registry(class_or_instance):
...
```
Then if I run tests I got this:
```
if cls._name in registry and not parents:
>           raise TypeError(
                "Component %r (in class %r) already exists. "
                "Consider using _inherit instead of _name "
                "or using a different _name." % (cls._name, cls)
            )
E           TypeError: Component 'component1' (in class <class 'odoo.addons.component.tests.test_build_component.TestBuildComponent.test_check_parent_abstract_over_component.<locals>.Component1'>) already exists. Consider using _inherit instead of _name or using a different _name.

odoo/external-src/connector/component/core.py:830: TypeError
```
I'm not an expert but think CleanUp wasn't called on set up test class.
I'm not an expert in that kind of stuff, so I try to add what I met in such code:
```
@class_or_instance.addCleanup
        def _close_and_roolback():
            env.cr.rollback()  # we shouldn't have to commit anything
            env.cr.close()
```
And this gives me:
```
@class_or_instance.addCleanup
>       def _close_and_roolback():
E       TypeError: addCleanup() missing 1 required positional argument: 'function'
```
Can you help me pls?

Reply:
_close_and_rollback is not here anymore as I asked you to rollback your changes and keep the old implementation. Have you tried again after that?